### ISSUE TITLE
[Feature request] Create folder when using the `output` parameter

### ISSUE THREAD

**Issue by michaelmass** [princjef/gomarkdoc#35]
When using the parameter `-output` with a template like `doc/{{.Dir}}/README.md`, the cli returns an error if the folder doesn't exist.
```
failed to write output file doc\pkg\foo\README.md: open doc\pkg\foo\README.md: The system cannot find the path specified.
```

Maybe the cli should automatically create the folder or it could be a new flag `-create-dirs`.

I have already made a fork for it. If it is accepted, I will open a PR:
https://github.com/michaelmass/gomarkdoc/commit/b868c0fc4a6a07512eaf94f072f4325ba6efc709

**Comment by princjef** [comment]
Yeah that's a reasonable addition. It seems simpler to me to have it be the default behavior rather than a flag, unless you know of use cases where people _wouldn't_ want that behavior.

I took a quick look at your change. Feel free to open a PR and we can get this merged/published.

**Comment by princjef** [comment]
Closed by #36
